conans/model/build_info.py

```diff
@@ -185,7 +186,7 @@ class CppInfo(_CppInfo):
     Defined in user CONANFILE, directories are relative at user definition time
     """
 
-    def __init__(self, ref_name, root_folder):
+    def __init__(self, ref_name, root_folder, version):
```

Comment:
I believe this change is not needed

Reply:
@danimtb how can we populate `self.components` in https://github.com/conan-io/conan/pull/7524/files#diff-068473d8621680eca1f2101d6be5ea0aR201 if we do not take the version as parameter here ?

Reply:
the `version` attribute is populated later, so you can add the `self.version` attribute without value and avoid changing the constructor of the class